JASP-Engine/JASP/R/descriptives.R

```diff
@@ -184,6 +184,34 @@ Descriptives <- function(jaspResults, dataset, options) {
     }
   }
 
+
+#X bar Chart
+if (options$Xbarchart){
+
+  Xbarchart <- jaspResults[["Xbarchart"]]
+
+  for (var in variables){
+    if(is.null(Xbarchart[[var]]) && .descriptivesIsNumericColumn(dataset.factors, var)){
+        Xbarchart[[var]] <- .Xbarchart(jaspResults = jaspResults, dataset= dataset, options= options, variables= var)
+        jaspResults[["Xbarchart"]]$dependOn(c("Xbarchart", "variables", "Rchart"))
+    }
+  }
+}
```

Comment:
So there's a couple of issues here. 

1. `if(is.null(Xbarchart[[var]])` -- you also do this check within the function `.Xbarchart`, no need to do it twice.
2. you never defined `jaspResults[["Xbarchart"]]` to be a collection. As a consequence, JASP only shows a single plot even though (run the analysis with multiple variables).
3. `Xbarchart` depends on an unrelated option, "Rchart".

I'd do something like:
```r
if (is.null(jaspResults[["Xbarchart"]])) {
  jaspResults[["Xbarchart"]] <- createJaspCollection(title = "...")
  jaspResults[["Xbarchart"]]$dependOn("Xbarchart")
}

Xbarchart <- jaspResults[["Xbarchart"]]
for (var in variables){
    if(.descriptivesIsNumericColumn(dataset.factors, var)) {
        Xbarchart[[var]] <- .Xbarchart(jaspResults = jaspResults, dataset= dataset, options= options, variables= var)
        Xbarchart[[var]]$dependOn(optionContainsValue=list(variables=var))    
  }
}
```
the same goes for the Rcharts.